src/include/session.h

```diff
@@ -30,6 +30,23 @@ struct __wt_hazard {
 #endif
 };
 
+/*
+ * WT_HAZARD_ARRAY --
+ *   An array of all hazard pointers held by the session.
+ *   New hazard pointers are added on a first-fit basis, and on removal their entry
+ *   in the array is set to null. As such this array may contain holes.
+ */
+struct __wt_hazard_array {
+/* The hazard pointer array grows as necessary, initialize with 250 slots. */
+#define WT_SESSION_INITIAL_HAZARD_SLOTS 250
+
+    wt_shared WT_HAZARD *arr;  /* The hazard pointer array */
+    wt_shared uint32_t inuse;  /* Number of array slots potentially in-use. We only need to iterate
+                                  this many slots to find all active pointers */
+    wt_shared uint32_t active; /* Number of array slots containing an active hazard pointer */
```

Comment:
May be active_size?

Reply:
`active` is the number of hazard pointers actually protecting a ref so I'm not sure size works here. I'm worried these names are a bit convoluted though so happy to discuss better ones.

Reply:
active_num?

Reply:
Fair point. I've renamed to `num_active` in [6ecdb33](https://github.com/wiredtiger/wiredtiger/pull/9782/commits/6ecdb33ca41e95348d9fbb79e0f11f9d52959c74)